cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -195,4 +197,8 @@ func (kc *kubeletConfig) Default(cfg *kubeadmapi.ClusterConfiguration, _ *kubead
 		klog.V(1).Infof("the value of KubeletConfiguration.cgroupDriver is empty; setting it to %q", constants.CgroupDriverSystemd)
 		kc.config.CgroupDriver = constants.CgroupDriverSystemd
 	}
+
+	if features.Enabled(cfg.FeatureGates, features.NodeLocalCRISocket) {
+		kc.config.ContainerRuntimeEndpoint = nodeRegOpts.CRISocket
```

Comment:
this `kc` is the global config that will be uploaded to the cluster.
i don't think we want to put the local node CRI socket on it.

Reply:
I checked the init/join flow carefully, and you are right, we should not upload it to the global configuration, but the **Design Details** section of the KEP clearly states that we will do so. I will modify the code and modify the description in the KEP after this PR is merged.